### DISCUSSION TITLE
question : watch sts ,watch not working as expected

### DISCUSSION THREAD

**Discussion by digeler** [kubernetes-client/csharp#875] [Q&A]
using aks 1.16.10 trying to watch statefullset replica but not getting any reply 
namespace watch
{
    internal class Program
    {
       private static void Main(string[] args)
            {
                var config = KubernetesClientConfiguration.BuildConfigFromConfigFile();

                IKubernetes client = new Kubernetes(config);

                var sts = client.ListNamespacedStatefulSetWithHttpMessagesAsync("default", watch: true, labelSelector: "test=app");
           using (sts.Watch<V1StatefulSet, V1StatefulSetList>(
               (type, item) =>
               {
                   if (type == WatchEventType.Modified)
                   {

                       {
                           Console.WriteLine(item.Metadata.ClusterName);
                           Console.WriteLine(WatchEventType.Added);
                       }
                   }
               },
               e => Console.WriteLine(e.Message)))
           {

           }
            }
        }
    }
is there anything am missing here ,thanks

**Comment by brendandburns** [comment]
The code looks correct. Are you not seeing events when a stateful set changes?

Can you send instructions for reproducing the bug that you are seeing?

Thanks

**Comment by digeler** [comment]
Thanks Brendan ,
i have researched some more, seems like my selector was not set correctly here is the working sample  : 

  internal class Program
    {
        private static void Main(string[] args)
        {
            var config = KubernetesClientConfiguration.BuildConfigFromConfigFile();

            IKubernetes client = new Kubernetes(config);

            var sts = client.ListStatefulSetForAllNamespacesWithHttpMessagesAsync(watch:true,fieldSelector: "metadata.name=typetagger");
            using (sts.Watch<V1StatefulSet,V1StatefulSetList>((type, item) =>
            {
                Console.WriteLine(item.Spec.Replicas);
            }))
            {
                Console.WriteLine("press ctrl + c to stop watching");

                var ctrlc = new ManualResetEventSlim(false);
                Console.CancelKeyPress += (sender, eventArgs) => ctrlc.Set();
                ctrlc.Wait();
            }
        }
    }
thank you so much for the help.
and sorry for hitting the fire button to soon :)

**Comment by brendandburns** [comment]
Cool, I'm glad it's working. Closing this issue.
